### DISCUSSION TITLE
Is it possible to lower a cluster member on index error?

### DISCUSSION THREAD

**Discussion by Gomez12** [ravendb/ravendb#16810] [Q&A]
Today we had the following situation : 1 node of 3 had an indexing error on a certain index.
But this resulted in that our application started giving error 500's as it uses that index and when the conditions where right that the application would hit that node.

This seems logical, but when we re-enabled the index again it lowered the node to rehabilitation status as the indexes were out of date.
Which results in no error 500's.

Isn't it more logical to put the node in rehabilitation (or at least as last priority) so on error people receive the least amount of errors? Or is there a config setting for it (I can't seem to find it)

**Comment by ayende** [comment]
The problem is detecting this kind of error and responding to it in time.
There are two types of scenarios here:

* This is a real indexing error (for example, divide by zero), which means that shortly all indexes in the cluster will have this error
* This is an error on one machine  (for example, run out of disk space)

This gets more complex when you may have _multiple_ distinct set of failed indexes.
